source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -159,6 +159,11 @@ void ShaderGraph::createConnectedNodes(const ElementPtr& downstreamElement,
 
 void ShaderGraph::addUpstreamDependencies(const Element& root, ConstMaterialPtr material, GenContext& context)
 {
+    if (!context.getOptions().addUpstreamDependencies)
+    {
+        return;
```

Comment:
Maybe do this check before calling the method instead. It's called in only two places, and one of them is in a loop over outputs, so would be more efficient to avoid the loop completely.

Reply:
Wasn't sure what you preferred. I'll move it out.